[5.2] Fix activation status

Amitesh007z opened the pull request:
Pull Request for Issue #45063 .

### Summary of Changes
Fixed the activation status issue where the "Activated" flag remained inactive even after the user verified their email.

Ensured that once a user completes email verification, the administrator can see the updated activation status correctly unless and untill the New User Account Activation is not Adminstator , like if its Adminstator , once Admin approves by clicking the link in his mail ,it turns activated. 

Modified language/en-GB/ and administrator/language/en-GB/ files to reflect updated changes.



### Testing Instructions
1.Register aa a  new user and check if email verification updates the activation status.

2.Test cases where New User Account Activation is set to different values (e.g., Self, Admin) to ensure correct behavior i.e if self ,then on clicking link updates activated in admin panel and for Admin , changes to activated in admin panel upon approval of admin by clicking link in mail received upon mail verification of user.

3.Verify that the updated language strings are correctly reflected in the UI.



### Actual result BEFORE applying this Pull Request
![image](https://github.com/user-attachments/assets/0233ba9e-1ee7-45f6-8d06-76e7d97699bb)




### Expected result AFTER applying this Pull Request
Beforre verification :
![image](https://github.com/user-attachments/assets/c2e8fabc-4669-4f36-a8d7-8830037b6d74)

Once after verification also by admin(as it is set as By Admin instead of self):
![image](https://github.com/user-attachments/assets/2c328b67-0b8f-4d59-b158-788605258d17)

Once after language change:
![image](https://github.com/user-attachments/assets/fc548ec8-b740-4532-ad43-5ce981f80e00)





### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

Commit:
Fix activation status issue #45063

Commit:
Fix activation status issue #45063

Commit:
Updated English language files

brianteeman commented:
For the reasons I stated in the original issue report I am personally not in favour of changing the string to "verified" as in some of the user registration methods there is no verification step.

Before even looking at changing anything I think you need to create a grid showing the values for enabled and activated at each step of the registration process for each scenario. Only then can we even determine if something is not working correctly. These are some of the oldest settings in Joomla and I would be very careful changing anything without having a full understanding of the intended results

fgsw commented:
@Amitesh007z Can you please:

- append the issue number (45063) in "Pull Request for Issue #."
- start the title "[5.3] " as titles are used in the changelog.

Thank you.

Amitesh007z commented:
@fgsw sorry for the mistake , i have edited and changed , 
Thanks

Commit:
Fix activation status issue #45063

Commit:
Fix activation status issue #45063

fgsw commented:
@Amitesh007z Thanks for changing the title. Can you please delete the issue number from the title as it makes no sense to have it in the changelog.

richard67 commented on administrator/components/com_users/src/Model/UserModel.php:
```diff
@@ -506,8 +506,11 @@ public function activate(&$pks)
+                        $table->block      = 0;
+                        $table->activation = '';
+                    }
+                    
```
```suggestion
```
Fix PHPCS

Amitesh007z replied:
> Fix PHPCS

Resolved the issue ,
sorry for the mistake
Thanks

Commit:
Fix PHPCS issue (no_whitespace_in_blank_line)

Commit:
Fix PHPCS issue (no_whitespace_in_blank_line)

Amitesh007z commented:
> @Amitesh007z Thanks for changing the title. Can you please delete the issue number from the title as it makes no sense to have it in the changelog.

yeah done , sorry for inconvenience caused , 
thanks

richard67 commented on administrator/components/com_banners/src/Table/BannerTable.php:
```diff
@@ -57,7 +57,6 @@ public function __construct(DatabaseDriver $db, ?DispatcherInterface $dispatcher
 
         $this->created = Factory::getDate()->toSql();
         $this->setColumnAlias('published', 'state');
-        $this->setColumnAlias('title', 'name');
```
Sorry, but removing this line of code seems not related to the purpose of this PR, and it reverts this PR: #44864 .
So that seems to be a mistake.

Amitesh007z replied:
Apologies for the unintended change, @richard67 . I’m surprised to see this as I only worked on the specific areas related to the issue and not on this file . I’ll review and restore the removed line to ensure it doesn’t affect PR #44864. I appreciate your feedback and will request a review again once all the fixes are in place. Thanks !

richard67 commented:
This PR shows a lot of unrelated changes, e.g. fine administrator/language/en-GB/install.xml and other XML files, composer.json and so on. Maybe the base branch was chosen wrong?

Commit:
fix user message

Amitesh007z commented:
> This PR shows a lot of unrelated changes, e.g. fine administrator/language/en-GB/install.xml and other XML files, composer.json and so on. Maybe the base branch was chosen wrong?

Sorry for the inconvenience caused till now @richard67 , i have changed only few set of lines which are related to the issue i.e state change of activated and enabled basically , also as an  additional task was to rename and change few terms with better replacements to avoid confusions and in order to provide clarity ,so i have changed some lines in language/en-GB mainly , will look into all errors, in CI tests and once everything is sorted will be happy to hear your review ,Sorry for calling upon re-review without checking all cases in CI tests passed,
Thanks

richard67 commented:
I think the problem is that your fix-activation-status branch for this PR was created based on a different branch than 5.2-dev (maybe 5.3-dev?) and then this PR was created for 5.2-dev, so the branches do not match and the PR shows all the unrelated changes because it shows the differences of the 5.2-dev and 5.3-dev branches, too. You could try to rebase this PR to 5.3-dev, or you could make a new PR.

Commit:
fix user message

Amitesh007z commented:
> I think the problem is that your fix-activation-status branch for this PR was created based on a different branch than 5.2-dev (maybe 5.3-dev?) and then this PR was created for 5.2-dev, so the branches do not match and the PR shows all the unrelated changes because it shows the differences of the 5.2-dev and 5.3-dev branches, too. You could try to rebase this PR to 5.3-dev, or you could make a new PR.

I have reviewed the branch sir , and it appears to be in sync with 5.2-dev. However, I will cross-check once more to ensure there are no discrepancies. If I find any inconsistencies, I will take the necessary steps to resolve them and update accordingly. Thank you for your guidance sir !

Commit:
Fixed PHPStan issues by replacing deprecated methods and classes.

Commit:
Fixed PHPStan issues by replacing deprecated methods and classes.

Amitesh007z commented:
@richard67 

I noticed that my current PR includes several unintended changes. To keep the contribution clean and focused, I will close the existing PR and create a new one with only the necessary modifications. will get it done in couple of hours with intended chnages Thanks
